crates/core/src/entry/validation_dependencies.rs

```diff
@@ -0,0 +1,115 @@
+use crate::network::entry_with_header::EntryWithHeader;
+use holochain_core_types::entry::Entry;
+use holochain_persistence_api::cas::content::Address;
+
+pub trait ValidationDependencies {
+    fn get_validation_dependencies(&self) -> Vec<Address>;
+}
+
+impl ValidationDependencies for EntryWithHeader {
+    fn get_validation_dependencies(&self) -> Vec<Address> {
+        match &self.entry {
+            Entry::App(_, _) => {
+                // in the future an entry should be dependent on its header but
```

Comment:
Just a note, in the future it could also include a user-defined pure function of the entry which returns extra dependencies to require for validation